### DISCUSSION TITLE
How to access github actions log of current run from a later step/job

### DISCUSSION THREAD

**Discussion by david-gang** [community/community#27097] [Actions]
<p>In jenkins it is possible to access the data of the current running job through groovy: <a href="https://medium.com/@elhayefrat/access-to-a-pipeline-jobs-own-jenkins-console-log-6619aec9a2be" class="inline-onebox" rel="noopener nofollow ugc">access to a pipeline job’s own Jenkins console log | by elhay efrat | Medium</a><br>
How is this possible in github actions?</p>
<p>My use case is that i have a job which is compiling and running unit tests. I want to take after a failure the console log, parse it and extract the root cause of the failure. I could pipe all output to a log file but then the console log. would remain empty which is not nice. so what would be the best way to accomplish it?</p>
<p>I could do it in a job which is triggered after any job run but i would look for an inclusive solution</p>

**Comment by ahdbilal** [comment]
<p>Hi! You can use GitHub CLI or Actions API to do that. CLI is the easiest to start with. Here is an <a href="https://github.blog/2021-04-15-work-with-github-actions-in-your-terminal-with-github-cli/">article</a> that can provide more information on how to set up and use GitHub CLI.</p>

**Comment by david-gang** [comment]
<p>Thanks for your quick reply. I read the article but still don’t understand how  the cli solves the problem</p>
<p>Let’s assume this hypothetical workflow:</p>
<pre><code class="lang-auto">- name: Gradle clean
   run: ./gradlew build
- name: Analyze error
  if: ${{failure()}}
  run: python3 ./analyze_and_slack_error.py $JOB_CONSOLE_FILE
</code></pre>
<p>So i want to run gradle build and if it fails i want to take the console output and analyze it with a python script i write</p>
<p>we don’t have the JOB_CONSOLE_FILE environment variable. Can you please help me how to use the github cli to get the console output</p>
<p>What should be the bash command here ?</p>
<pre><code class="lang-auto">export JOB_CONSOLE_FILE = `gh `
</code></pre>

**Comment by jsoref** [comment]
<p>In theory,</p>
<p><a href="https://docs.github.com/en/rest/reference/actions#get-a-workflow-run" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/actions#get-a-workflow-run</a></p>
<p>Will offer:</p>
<pre><code class="lang-auto">  "logs_url": "https://api.github.com/repos/octo-org/octo-repo/actions/runs/30433642/logs",
</code></pre>
<p>You can also use <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs" rel="noopener nofollow ugc">outputs</a> in case you need to send data between workflows.</p>
<p>Personally, for the <em>basic</em> use case you’re describing, I’d just use <a href="https://man7.org/linux/man-pages/man1/tee.1.html" rel="noopener nofollow ugc">tee</a> in the first step (sending the output to a file) and then parse the captured output in the second step.</p>

**Comment by jsoref** [comment, marked as answer]
<p>Sample (not actually tested; always validate scripts before randomly running them in your own trusted environment; …):</p>
<pre><code class="lang-auto">- name: Gradle clean
   run: (./gradlew build 2&gt;&amp;1) | tee /tmp/gradle.out
- name: Analyze error
  if: failure()
  run: python3 ./analyze_and_slack_error.py /tmp/gradle.out
</code></pre>
